### ISSUE TITLE
Loading Sounds into WAV BANK Crashes Rack

### ISSUE THREAD

**Issue by jonheal** [clone45/voxglitch#32]
Rack 1.1.6
Voxglitch 1.6.0
Windows 10

I grabbed a random group of sounds from my library and put them in a folder. The total size of the folder was 246MB. Varying file sizes from 731KB to 34MB.

Upon identifying the folder from WAV BANK's context menu, I get the spinny wheel for about five seconds and then Rack barfs.

Once I deleted the largest files from the folder, WAV BANK appeared to successfully load them, but then several seconds later, Rack crashed again. Total folder size this time: 58MB

**Comment by clone45** [comment]
@jonheal Thank you so much for the detailed bug report.  Looks like I'll need to do some serious work on that module to be able to handle lots of large audio files. At the moment, it attempts to load all of the samples into memory, which is no-doubt why it's crashing.  I'll try to modify the module to stream the audio instead.  It could take me a few days to start working on this.  Sorry again for the issues.

**Comment by jonheal** [comment]
I have 8GB of memory on this machine, so there should be plenty of RAM for the audio files I selected. Maybe some memory pointers are stepping on each other? Also, I notice it takes a bit of time for samples to start playing once selected, so it feels like they are not being retained in RAM, but reloaded with each selection. I don’t know if streaming from the disk would make that speedier.

**Comment by jonheal** [comment]
Loading and playing multiple, long WAV files vastly improved in v.1.10.0. THANKS!

**Comment by bret-2600** [comment]
@jonheal I'm very sorry that I haven't had time to address this issue yet.  I didn't do anything in version 1.10.0 related to this, so I'm surprised that the issues improved.

I keep waiting for an "ah ha!" moment to strike.  I researching the ability to stream .wav files but didn't find any tangible leads via Google.  I haven't delved deeply into memory usage in VCV Rack, which is probably what I need to do, but just haven't taken the time yet.  It's still on my radar and I'll try to research it soon.

**Comment by jonheal** [comment]
What improved is I was dumb, and suddenly realized some the folders of files I was trying to load were full of AIF files, not WAV files. A folder of WAV files loads and plays as expected — until it hits some limit in either individual file size of collective file size. But … appreciate your modules nonetheless. 😊

**Comment by napstablup** [comment]
Hello
so i was trying to load a wav file ( it's only 4 seconds long, 1.38 MB ) and the 'sample' graph stays empty. Can you please help. 
also i loved your descriptions on some of the plugins, can't wait to use them. Thanks

**Comment by clone45** [comment]
Sure!  Can you send the .wav file to me?

**Comment by napstablup** [comment]
Yeah, sure
check your mail )

**Comment by clone45** [comment]
@napstablup There seems to be a space in the filename that's causing that sample not to load.  The file I received from you is called  `SFX_Freeze. wav` Could you try removing the space between the period and the extension and see if it starts working for you?

**Comment by napstablup** [comment]
Thanks, i will try tomorrow. Hope it helps. I will write you later about the result.  воскресенье, 05 апреля 2020г., 19:53 +03:00 от Bret Truchan  notifications@github.com :

**Comment by napstablup** [comment]
so i've changed the name to just 'freeze.wav' and tried to load the file, but it didn't work. still empty in the 'sample' graph

**Comment by clone45** [comment]
Thanks for trying that!  Hmm..  are there any other .wav files in that folder?  Wav Bank will try and load all .wav files in a folder.  If the folder has a lot of files, it might crash, which is a known issue.  Let me know.  I'll keep helping until we figure it out!

**Comment by napstablup** [comment]
Thanks for your help. I've only tried to load this one in 'ghost'. Ive also tried to load different length files into 'goblins' and 'bank'. They all react the same (

**Comment by clone45** [comment]
@napstablup It seems to be working for me.  I've created a video showing it working in Ghosts: https://youtu.be/VaCS0OFrJE0.  Take a peek and let me know what you think.

**Comment by napstablup** [comment]
Thanks, i will let you know

**Comment by napstablup** [comment]
yeah, so i've just loged out and loged back in and tried this method and it worked ! thanks a lot for your help and your time

**Comment by clone45** [comment]
My pleasure!  Happy to help.
